eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/NestedBundle.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.Keys;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.NullSafe;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class NestedBundle extends AbstractCachingOptional<Bundle>
+{
+    public NestedBundle(final Intent intent)
+    {
+        super(new Factory<Optional<Bundle>>()
+        {
+            @Override
+            public Optional<Bundle> create()
+            {
+                return new NullSafe<>(intent.getBundleExtra(Keys.NESTED_BUNDLE));
```

Comment:
I thought about this `Obligatory` in the weekend and now I think that it can be misleading to use. I mean that wherever we need an `Obligatory<T>`, don't we actually need a `T`?
That's what its name means I think, so it's no longer an `Optional<T>`. If we pass it around as `Optional<T>` subsequent code will see it that way, providing its own defaults or routing on `isPresent()`, while it really should not at that point, because that is no longer a valid program flow.

As I see the motivation in this case, it would be to have a lazy `Bundle` that always evaluates to something. I think that's not the same as `Optional<Bundle>`.
Because `Bundle` is not an interface we can't do it in an implementation, so I guess the alternative to
```
Bundle bundle = new NestedBundle(getIntent()).value(null)
or
Bundle bundle = new NestedBundle(getIntent()).value(new Bundle());
```
would be a factory then..  but I am not sure that's worth it.

The contract that nested bundle would always present be - would than mean we also make the contract that all Fragments has a non-null argument?
The null check in `Argument` is because of that, so that `fragment.getArguments()` can be null, so just to be convenient to use `new Arugment(.. fragment.getArguments()). 
We could set up these contracts to eliminate nulls as much as possible, then we need to keep to that, making sure to add empty bundles in the parameterless Fragment factory methods.

Let me know if we should do this.